include/highfive/bits/H5DataType_misc.hpp

```diff
@@ -169,31 +169,14 @@ class AtomicType<FixedLenStringArray<StrLen>>: public DataType {
         : DataType(create_string(StrLen)) {}
 };
 
-template <>
-inline AtomicType<std::complex<float>>::AtomicType() {
-    static struct ComplexType: public Object {
-        ComplexType() {
-            _hid = H5Tcreate(H5T_COMPOUND, sizeof(std::complex<float>));
-            // h5py/numpy compatible datatype
-            H5Tinsert(_hid, "r", 0, H5T_NATIVE_FLOAT);
-            H5Tinsert(_hid, "i", sizeof(float), H5T_NATIVE_FLOAT);
-        };
-    } complexType;
-    _hid = H5Tcopy(complexType.getId());
-}
-
-template <>
-inline AtomicType<std::complex<double>>::AtomicType() {
-    static struct ComplexType: public Object {
-        ComplexType() {
-            _hid = H5Tcreate(H5T_COMPOUND, sizeof(std::complex<double>));
-            // h5py/numpy compatible datatype
-            H5Tinsert(_hid, "r", 0, H5T_NATIVE_DOUBLE);
-            H5Tinsert(_hid, "i", sizeof(double), H5T_NATIVE_DOUBLE);
-        };
-    } complexType;
-    _hid = H5Tcopy(complexType.getId());
-}
+template <typename T>
+class AtomicType<std::complex<T>>: public DataType {
+  public:
```

Comment:
what about `static_assert(std::is_arithmetic<T>, )` ?

Reply:
is_arithmetic is not enough, it should be is floating point.
But I think it is one more check for nothing because the user should not write std::complex of smth that is floating point. His program will be ill-formed.

Reply:
It's a "good to have". Not a blocker